### DISCUSSION TITLE
Disable move mapping for specific language

### DISCUSSION THREAD

**Discussion by jborbik** [nvim-treesitter/nvim-treesitter-textobjects#422] [Q&A] [answered]
Hello,
Is there a way to disable move mapping for a specific language, i.e. enable it only conditionally? I would like to disable the function move keymaps (`]m` etc.) for cpp.

**Comment by kiyoon** [comment, marked as answer]
You'll need to manually map those keymaps. You can use `:TSTextobjectGotoNextStart @function.outer`

**Comment by jborbik** [comment]
Great, that helps! Anyway, do you know why `@function.outer` is set on lines where template variables are defined? E.g. it would stop on:
```
  Eigen::Transform<double,3,Eigen::Isometry> test(Eigen::Isometry3d::Identity());
```
Isn't that a bug?

**Reply by kiyoon** [reply]
Is that a variable not a function? It could be a bug, in which case you'll need to modify the cpp queries to not select those

**Reply by jborbik** [reply]
Thank you once more!

**Reply by kiyoon** [reply]
I haven't used C++ for a long time, and it's better if someone with more experience contributes to writing queries. If you're interested, it's not too difficult. You need to install `nvim-treesitter/playground` and open it with `:TSPlaygroundToggle`. Press `o` and it will open a buffer to put queries in. Paste the queries from this repo `queries/cpp/textobjects.scm`. Then you'll see which parts of your code are highlighted with the queries. You can tune the wrong queries (or write new queries if you want) and create a PR.

If you're not interested, you can also make an issue and describe in more detail what the problem is. Someone else may help fixing the issues.

**Reply by jborbik** [reply]
Wow, that's very useful. I'll play with it and create the PR after I leave the playground ;) Thank you for the patience and help!

**Reply by kiyoon** [reply]
No worries, any contribution will be much appreciated!

**Reply by jborbik** [reply]
I couldn't fix it by myself - I am not proficient with lisp queries. I have opened an issue here: https://github.com/nvim-treesitter/nvim-treesitter/issues/4552 I still learned something about it, so thank you a lot! Hope opening the issue still helps :)

**Reply by kiyoon** [reply]
I initially thought it could be the query bug, but reading through the issue it looks like a parser bug as mentioned in the comment. If the parser says that's function declaration when it's clearly not, there's nothing much we can do with queries. Anyways thanks for discovering the issue!
